src/Identity/IdentityServer/UserDecryptionOptionsBuilder.cs

```diff
@@ -130,6 +133,12 @@ private async Task BuildTrustedDeviceOptions()
             hasManageResetPasswordPermission = await _currentContext.ManageResetPassword(_ssoConfig!.OrganizationId);
         }
 
+        // If tde was disabled the user can reset master pw
+        if (_user != null && !_user.HasMasterPassword() && _device != null && _device.IsTrusted())
+        {
+            hasManageResetPasswordPermission = true;
```

Comment:
We may want a different flag. The client needs to communicate this flow differently from the TDE offboarding one.

Reply:
Yeah, that makes sense. It does seem to work correctly this way though, so at least the flow/setting the keys seems correct. Just need to make it more robust in the code.

Reply:
Added "isTdeOffboarding" in 89979ad6eaee1304dc8a7a276d61945b8e7490f9. Needs client support aswell, but I think this should separate the cases clearly.

Reply:
Also event logging this change now.